clients/js/test/plugins/asset/updateDelegate.test.ts

```diff
@@ -256,3 +407,283 @@ test('it updateDelegate cannot update an asset after delegate authority revoked'
     uri: 'https://short.com',
   });
 });
+
+test('it cannot add additional delegate as additional delegate', async (t) => {
+  const umi = await createUmi();
+  const updateDelegate = generateSigner(umi);
+  const updateDelegate2 = generateSigner(umi);
+  const asset = await createAsset(umi, {
+    plugins: [
+      {
+        type: 'UpdateDelegate',
+        additionalDelegates: [updateDelegate.publicKey],
+      },
+    ],
+  });
+
+  const result = updatePlugin(umi, {
+    asset: asset.publicKey,
+    plugin: {
+      type: 'UpdateDelegate',
+      additionalDelegates: [
+        updateDelegate.publicKey,
+        updateDelegate2.publicKey,
+      ],
+    },
+    authority: updateDelegate,
+  }).sendAndConfirm(umi);
+
+  await t.throwsAsync(result, { name: 'NoApprovals' });
+});
+
+test('it can remove additional delegate as additional delegate if self', async (t) => {
+  const umi = await createUmi();
+  const updateDelegate = generateSigner(umi);
+  const updateDelegate2 = generateSigner(umi);
+  const asset = await createAsset(umi, {
+    plugins: [
+      {
+        type: 'UpdateDelegate',
+        additionalDelegates: [
+          updateDelegate.publicKey,
+          updateDelegate2.publicKey,
+        ],
+      },
+    ],
+  });
+
+  await updatePlugin(umi, {
+    asset: asset.publicKey,
+    plugin: {
+      type: 'UpdateDelegate',
+      additionalDelegates: [updateDelegate2.publicKey],
+    },
+    authority: updateDelegate,
+  }).sendAndConfirm(umi);
+
+  await assertAsset(t, umi, {
+    asset: asset.publicKey,
+    owner: umi.identity.publicKey,
+    updateAuthority: { type: 'Address', address: umi.identity.publicKey },
+    updateDelegate: {
+      authority: {
+        type: 'UpdateAuthority',
+      },
+      additionalDelegates: [updateDelegate2.publicKey],
+    },
+  });
+});
+
+test('it cannot remove another additional delegate as additional delegate', async (t) => {
```

Comment:
Do you have a test that they can remove themselves as an additional delegate?

Reply:
I think that's this, right?
'it can remove additional delegate as additional delegate if self'